daml/Marketplace/Distribution/Auction/Utils.daml

```diff
@@ -0,0 +1,77 @@
+module Marketplace.Distribution.Auction.Utils where
+
+import DA.Finance.Asset (AssetDeposit, AssetDeposit_Split(..))
+import Marketplace.Settlement qualified as Settlement
+import Marketplace.Distribution.Auction.Model qualified as Auction
+import Marketplace.Distribution.Bid.Model qualified as Bid
+import Marketplace.Distribution.Bid.Model (Bid(..))
+import Marketplace.Distribution.Bid.Service qualified as Bid
+import Marketplace.Trading.AllocationAccountRule qualified as Allocation
+import Marketplace.Trading.AllocationAccountRule (AllocationAccountRule(..))
+
+-- | Takes a list and splits it based off a predicate. Values which match the predicate go in the right list of the tuple
+splitList : (a -> Bool) -> [a] -> ([a], [a])
+splitList p xs = foldr (\x (y,z) -> if p x then (y, x :: z) else (x :: y, z)) ([],[]) xs
+
+-- | Returns a bidder's allocation to their trading account
+withdrawAllocation : Bid.T -> Update (ContractId AssetDeposit)
+withdrawAllocation bid@Bid{..} = do
+    exerciseByKey @AllocationAccountRule allocationAccount.id Allocation.Withdraw with transferTo = tradingAccount; ..
+
+-- | Updates a bid's status
+updateBidStatus : Bid.Status -> Bid.T -> Update (ContractId Bid)
+updateBidStatus newStatus bid@Bid{..} = do
+    exerciseByKey @Bid.S (bid.assetId, bid.customer) Bid.UpdateStatus with ..
```

Comment:
I've seen this pattern `with ..` used a lot in the marketplace, especially for Request/Propose patterns. I'm wondering whether this is in fact an anti-pattern, the alternative being passing a nested structure. i.e. instead of

```
template Proposal with a : A, b : B, c : C

Propose : Proposal with a, b, c do create Proposal ...
```
it might be better to use
```
template Proposal with result : Result
template Result with a: A, b : B, c : C

propose: Proposal with result: Result
```
I even remember seeing a related post about this in the forum; I'll try and find it.

The downside of the `..` aproach is that you end up having to create these one-line wrapper functions, instead of calling the choices directly. I'm also not sure if the compiler is smart enough to avoid de/re-constructing the data types, leading to overhead.

(Sorry, not picking on you here particularly - it's a wider thing which I think we should consider changing CCing @GeorgSchneider )

Reply:
Yeah, here you go: https://discuss.daml.com/t/maximum-number-of-record-fields/2073/3

Reply:
Issue has added to the list of discuss points for the overall Marketplace project